src/ray/gcs/redis_context.cc

```diff
@@ -183,11 +188,16 @@ const std::string &CallbackReply::ReadAsPubsubData() const {
 
 size_t CallbackReply::ReadAsScanArray(std::vector<std::string> *array) const {
   RAY_CHECK(reply_type_ == REDIS_REPLY_ARRAY) << "Unexpected type: " << reply_type_;
-  *array = string_array_reply_;
+  array->clear();
+  array->reserve(string_array_reply_.size());
+  for (const auto &element : string_array_reply_) {
+    RAY_CHECK(element.has_value());
+    array->emplace_back(*element);
+  }
   return next_scan_cursor_reply_;
 }
 
-const std::vector<std::string> &CallbackReply::ReadAsStringArray() const {
+const std::vector<std::optional<std::string>> &CallbackReply::ReadAsStringArray() const {
```

Comment:
This change requires `std=c++17` and breaks my local build for MSVC on windows. Where does one specify to bazel that c++17 is required?

Reply:
Hi @mattip, we have upgraded to c++17 recently: https://github.com/ray-project/ray/pull/18622/files

Reply:
Hmm. I wonder why building with `pip wheel .\python --use-feature=in-tree-build -vv` does not pick those changes up?

Reply:
I'm actually not familiar with that part.  Can you open an issue if you see a problem? Thanks.

Reply:
#19070